org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ClasspathUpdateHandler.java

```diff
@@ -54,7 +54,7 @@ public void elementChanged(ElementChangedEvent event) {
 		if (connection != null && uris != null && !uris.isEmpty()) {
 			for (String uri : uris) {
 				PreferenceManager preferenceManager = JavaLanguageServerPlugin.getPreferencesManager();
-				if (preferenceManager.getPreferences().isAnnotationNullAnalysisEnabled()) {
+				if (preferenceManager.getPreferences().isAnnotationNullAnalysisEnabled() && preferenceManager.getPreferences().isAutobuildEnabled()) {
```

Comment:
This is too earlier to skip. It's better to skip rebuild after `preferenceManager.getPreferences().updateAnnotationNullAnalysisOptions(javaProject)`.

Reply:
Good catch. changed in https://github.com/eclipse/eclipse.jdt.ls/pull/2228/commits/0cc470ff98b089dc115457c03d374e868fbea6f0